PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py

```diff
@@ -537,13 +537,18 @@ def _add_deepMET(process):
     process.load("RecoEgamma.EgammaTools.slimmedEgammaFromMultiCl_cff")
     phase2_hgcal.toModify(task, func=lambda t: t.add(process.slimmedEgammaFromMultiClTask))
 
-    # L1 pre-firing weights for 2016 and 2017
+    # L1 pre-firing weights for 2016, 2017, and 2018
     from Configuration.Eras.Modifier_run2_L1prefiring_cff import run2_L1prefiring
     from Configuration.Eras.Modifier_stage1L1Trigger_cff import stage1L1Trigger
     from Configuration.Eras.Modifier_stage2L1Trigger_2017_cff import stage2L1Trigger_2017
-    process.load("PhysicsTools.PatUtils.L1ECALPrefiringWeightProducer_cff")
-    stage1L1Trigger.toModify(process.prefiringweight, DataEra = "2016BtoH")
-    stage2L1Trigger_2017.toModify(process.prefiringweight, DataEra = "2017BtoF")
+    from Configuration.Eras.Modifier_stage2L1Trigger_2018_cff import stage2L1Trigger_2018
+    from Configuration.Eras.Modifier_tracker_apv_vfp30_2016_cff import tracker_apv_vfp30_2016
+    process.load("PhysicsTools.PatUtils.L1PrefiringWeightProducer_cff")
+    (stage1L1Trigger & tracker_apv_vfp30_2016).toModify(process.prefiringweight, DataEraECAL = "2016BtoH", DataEraMuon = "2016preVFP" )
+    (stage1L1Trigger & ~tracker_apv_vfp30_2016).toModify(process.prefiringweight, DataEraECAL = "2016BtoH", DataEraMuon = "2016postVFP" )
+    stage1L1Trigger.toModify(process.prefiringweight, DataEraECAL = "2016BtoH", DataEraMuon = "2016" )
```

Comment:
Doesn't this overwrite `DataEraMuon` of the previous two lines?

Reply:
True, I'll fix that.